src/kontrol/kdist/cheatcodes.md

```diff
@@ -907,6 +913,46 @@ Otherwise, throw an error for any other call to the Foundry contract.
       [owise]
 ```
 
+```k
+    rule [foundry.call.mockCall]:
+         <k> #call_foundry SELECTOR ARGS
+          => #loadAccount #asWord(#range(ARGS, 0, 32)) 
+          ~> #setMockCall #asWord(#range(ARGS, 0, 32)) #range(ARGS, #asWord(#range(ARGS, 32, 32)) +Int 32, #asWord(#range(ARGS, #asWord(#range(ARGS, 32, 32)), 32))) #range(ARGS, #asWord(#range(ARGS, 64, 32)) +Int 32, #asWord(#range(ARGS, #asWord(#range(ARGS, 64, 32)), 32)))
+         ...
+         </k> 
+      requires SELECTOR ==Int selector ( "mockCall(address,bytes,bytes)" )
+
+    //rule [foundry.call.mockCallWithMsgValue]:
+    //     <k> #call_foundry SELECTOR ARGS
+    //        => #loadAccount #asWord(#range(ARGS, 0, 32))
+    //        ~> #setMockCall #asWord(#range(ARGS, 0, 32)) #range(ARGS, 96, #asWord(#range(ARGS, 64, 32))) false ... </k>
+    //  requires SELECTOR ==Int selector ( "mockCall(address,uint256,bytes,bytes)" )
+```
+
+```k
+    rule [foundry.set.mockCall]:
+         <k> #next [ _OP:CallOp ] ~> (. =>  #setLocalMem RETSTART RETWIDTH RETURNDATA) ~> #execute ... </k>
+         <localMem> LM </localMem>
+         <wordStack> _ : ACCTTO : _ : ARGSTART : ARGWIDTH : RETSTART : RETWIDTH : _WS </wordStack>
```

Comment:
To avoid warnings, you can replace unused args with `_`, or prefix their name with underscore.
```suggestion
         <wordStack> _ : ACCTTO : _ : ARGSTART : _ : RETSTART : RETWIDTH : _WS </wordStack>
```